### PR TITLE
[@types/relay-runtime]: added getFragmentIdentifier

### PR DESCRIPTION
**Pull request by morrys**
This PR added [getFragmentIdentifier](https://github.com/facebook/relay/blob/v10.1.0/packages/relay-runtime/util/getFragmentIdentifier.js)
https://github.com/facebook/relay/blob/master/packages/relay-runtime/index.js#L324

```js
function getFragmentIdentifier(
  fragmentNode: ReaderFragment,
  fragmentRef: mixed,
): string

```

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

- [X] Provide a URL to documentation or source code which provides context for the suggested changes: 

- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## relay-runtime ([<kbd>unpkg</kbd>](https://unpkg.com/browse/relay-runtime@latest/))
was missing the following properties:
1. ReplaySubject
1. getNode
1. reportMissingRequiredFields
1. isFragment
1. isInlineDataFragment

<details>
<summary>as well as these 13 other properties...</summary>
<p>isRequest, MutationTypes, RangeOperations, MutationHandlers, VIEWER_ID, VIEWER_TYPE, PreloadableQueryRegistry, createPayloadFor3DField, RelayError, DEFAULT_HANDLE_KEY, isScalarAndEqual, recycleNodesInto, stableCopy</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against f4ee67ac700b14707ee8e4a709fd1e7e36bd4063
</p>

File: types/relay-runtime/lib/util/getFragmentIdentifier.d.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import { ReaderFragment } from './ReaderNode';
+
+export default function getFragmentIdentifier(fragmentNode: ReaderFragment, fragmentRef: any): string;
```
Comment (maraisr) [inline types/relay-runtime/lib/util/getFragmentIdentifier.d.ts:3]: Think there is a type for that fragmentRef: `FragmentRef`.

Comment (morrys) [reply types/relay-runtime/lib/util/getFragmentIdentifier.d.ts:3]: @maraisr Which do you mean?

Comment (morrys) [reply types/relay-runtime/lib/util/getFragmentIdentifier.d.ts:3]: @maraisr friendly ping

Comment (maraisr) [reply types/relay-runtime/lib/util/getFragmentIdentifier.d.ts:3]: Super sorry! This guy https://github.com/DefinitelyTyped/DefinitelyTyped/blob/903538301983042bd4ec105330e20e5a1a7f2a8a/types/relay-runtime/index.d.ts#L220

Comment (morrys) [reply types/relay-runtime/lib/util/getFragmentIdentifier.d.ts:3]: This is used only by readInlineData
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/903538301983042bd4ec105330e20e5a1a7f2a8a/types/relay-runtime/lib/store/readInlineData.d.ts#L4
and it was done as well as relay:
https://github.com/facebook/relay/blob/master/packages/relay-runtime/store/readInlineData.js#L27

While in this case relay used the mixed type

Comment (maraisr) [reply types/relay-runtime/lib/util/getFragmentIdentifier.d.ts:3]: relay's Flow types have been known to not always be accurate. Granted im not entirely sure what "getFragmentIdentifier" does, but to me it feels like the argument should be more strict than "any".

Comment (morrys) [reply types/relay-runtime/lib/util/getFragmentIdentifier.d.ts:3]: This is an internal function used in relay-experimental in order to calculate a unique fragment id.

I suppose that's why the fragmentRef typology has been relaxed.

on types/relay-runtime i would not deviate from their current implementation, at most you could propose a PR in relay and improve the typing and then report it here 👍

**Review by maraisr — APPROVED** [review]
LGTM

**Pull request merged**